### ISSUE TITLE
Question: Setting initiators in source

### ISSUE THREAD

**Issue by machadocs** [cle-b/httpdbg#147]
Hello,

Is there a way to explicitly set an initiator name in source? 

---

For context, I am using `httpdbg` in `pytest` and would like to have separate initiator groups for `fixtures` and `items` (tests) that make requests.
Currently, it seems that all requests made in setup/call/teardown phases are grouped under a single initiator for the `item`.

### Example

```python
@pytest.fixture(scope='session')
def loaded_stuff():
    return requests.get("/some/url").json()

def test_a(loaded_stuff):
    response = requests.post("/some/second/url", json=loaded_stuff)
    assert response.ok

def test_b(loaded_stuff):
    response = requests.post("/some/third/url", json=loaded_stuff)
    assert response.ok
```

The httpdbg output for the code above would show requests like so:

- `test_a`
  - `GET .../some/url`
  - `POST .../some/second/url`
- `test_b`
  - `POST .../some/third/url`

Note that requests made in the `loaded_stuff` fixture are grouped under the `test_a` initiator, while the `test_b` initiator omits requests made in `loaded_stuff` due to its scope. 

Ideally I would have the requests made in `loaded_stuff` under its own initiator - so is there a way I can set the initiator explicitly when inside a fixture?

**Comment by cle-b** [comment]
Hello :) 

No, it is not possible to add anything to your code to change the initiator when using `pytest`. 

I am working on a solution to include information about the pytest fixture in the UI.

**Comment by cle-b** [comment]
Hello @machadocs ,

I added information about the `pytest` fixture that initiated the HTTP requests.

This is available in [httpdbg v0.24.1](https://pypi.org/project/httpdbg/0.24.1/).

![image](https://github.com/user-attachments/assets/c75f362e-019a-422c-acb5-81f2b2857227)

I don't plan to make it possible to explicitly set information about the initiator in the code, so I hope this alternative will be helpful to you.

[documentation](https://httpdbg.readthedocs.io/en/latest/pytest/)

**Comment by machadocs** [comment]
Sorry for the delay, this is great. Thank you!
